Library/TestHelpers/TestCase.swift

```diff
@@ -71,11 +69,16 @@ internal class TestCase: FBSnapshotTestCase {
 }
 
 internal func preferredSimulatorCheck() {
```

Comment:
Can we keep this as:

```swift
internal func preferredSimulatorCheck() {
  guard
    let identifier = ProcessInfo().environment["SIMULATOR_MODEL_IDENTIFIER"],
    ["iPhone10,1", "iPhone10,4"].contains(identifier),
    #available(iOS 12.0, *)
  else {
    fatalError("Please only test and record screenshots on an iPhone 8 simulator running iOS 12")
  }
}
```
and then can we add this to `TestCase`:
```swift
override var recordMode: Bool {
  willSet(newValue) {
    if newValue {
      preferredSimulatorCheck()
    }
  }
}
```
This will only perform the `preferredSimulatorCheck()` when we are actually recording screenshots and not just running unit tests.
You can then remove the line calling `preferredSimulatorCheck()` from `setUp()` in `TestCase.swift` 👍